userspace/falco/falco_outputs.cpp

```diff
@@ -268,8 +276,21 @@ inline void falco_outputs::push(const ctrl_msg& cmsg)
 {
 	if (!m_queue.try_push(cmsg))
 	{
-		fprintf(stderr, "Fatal error: Output queue reached maximum capacity. Exiting.\n");
-		exit(EXIT_FAILURE);
+		switch (m_recovery)
+		{
+		case falco_common::RECOVERY_EXIT:
+			fprintf(stderr, "Fatal error: Output queue out of memory. Exiting ... \n");
+			exit(EXIT_FAILURE);
+		case falco_common::RECOVERY_EMPTY:
+			m_outputs_queue_num_drops += m_queue.size();
+			fprintf(stderr, "Output queue out of memory. Empty queue and continue ... \n");
+			m_queue.empty();
+			break;
+		default:
```

Comment:
I would use the `case falco_common::RECOVERY_CONTINUE:` here and the `default:` could become just an `exit(EXIT_FAILURE);` with a log message like `"strategy unknown"`. In this way we can check it will never happen, it's more future proof